src/pydvl/value/shapley/classwise.py

```diff
@@ -0,0 +1,251 @@
+"""
+Implementation of the algorithm footcite:t:`schoch_csshapley_2022`.
+"""
+import logging
+import numbers
+from concurrent.futures import FIRST_COMPLETED, wait
+from copy import copy
+from typing import cast
+
+import numpy as np
+
+from pydvl.utils import (
+    ParallelConfig,
+    Utility,
+    effective_n_jobs,
+    init_executor,
+    init_parallel_backend,
+)
+
+__all__ = [
+    "compute_classwise_shapley_values",
+]
+
+from tqdm import tqdm
+
+from pydvl.utils.score import ClasswiseScorer
+from pydvl.value.result import ValuationResult
+from pydvl.value.shapley.montecarlo import permutation_montecarlo_classwise_shapley
+from pydvl.value.shapley.truncated import TruncationPolicy
+from pydvl.value.stopping import MaxChecks, StoppingCriterion
+
+logger = logging.getLogger(__name__)
+
+
+def compute_classwise_shapley_values(
+    u: Utility,
+    *,
+    done: StoppingCriterion,
+    truncation: TruncationPolicy,
+    normalize_values: bool = True,
+    n_resample_complement_sets: int = 1,
+    use_default_scorer_value: bool = True,
+    min_elements_per_label: int = 1,
+    n_jobs: int = 1,
+    config: ParallelConfig = ParallelConfig(),
+    progress: bool = False,
+) -> ValuationResult:
+    """
+    Computes the classwise Shapley value by parallel processing. Independent workers
+    are spawned to process the data in parallel. Once the data is aggregated, the values
+    can be optionally normalized, depending on ``normalize_values``.
```

Comment:
The implementation details do not belong in the docsting. Instead we always document what the method computes.